src/main/java/com/amazon/opendistroforelasticsearch/sql/query/DefaultQueryAction.java

```diff
@@ -134,7 +134,7 @@ private void setIndicesAndTypes() {
      */
     public void setFields(List<Field> fields) throws SqlParseException {
 
-        if (!select.getFields().isEmpty()) {
+        if (!select.getFields().isEmpty() && !select.isSelectAll()) {
```

Comment:
Please add tests to cover this case.

Reply:
Based on Andy's recommendation, also add integ tests that have GROUP BY and ORDER BY to reduce the chance of breaking something else.

Reply:
If possible, please check JDBC format too.

Reply:
+1.